packages/ui/src/camel-utils/camel-to-table.adapter.ts

```diff
@@ -1,57 +1,131 @@
 import { IPropertiesRow, IPropertiesTable, PropertiesHeaders } from '../components/PropertiesModal';
-import { ICamelComponentDefinition, ICamelProcessorDefinition, IKameletDefinition } from '../models';
+import {
+  ICamelComponentApi,
+  ICamelComponentHeader,
+  ICamelComponentProperty,
+  ICamelProcessorProperty,
+  IKameletDefinition,
+} from '../models';
 
-//since ICamelComponentDefinition and share the same values, this table definition can be used for both
-export const camelComponentToTable = (
-  componentDef: ICamelComponentDefinition | ICamelProcessorDefinition,
+export type ICamelComponentCommonProperties = ICamelComponentProperty | ICamelComponentHeader;
+export type IPropertiesTableFilter = { filterKey: string; filterValue: string };
+
+const fullFillFilter = (value: any, filter: IPropertiesTableFilter): boolean => {
```

Comment:
[curious-question]: Is there a more specific type here instead of `any`?

Reply:
unfortunetelly not since it is used for all types of component, or i can define `export type` with all type of properties from modals files

Reply:
^  i remember that I have already tried this but I can use that since `keyof IFilterableProperties`  contains only keys which are in each property. So, for example, if I want to filter by `kind` key which is only in camel/processor properties, i cannot use that

Reply:
resolved via chat to use generics